### PR TITLE
Add option to pass dataCallback from the ember config

### PR DESCRIPTION
**Pull request by hussfelt**
This enables you to modify the stack-payload before passing it to Sentry.

Work in progress - cant verify success yet.

### CODE REVIEW

**Comment by damiencaselli** [comment]
Hi @hussfelt, thanks for the PR.

This would indeed be a nice addition, though I'm not a huge fan of the function definition inside the config file.

May be we could instead try to use a *sentry-data-callback* service that would have to be registered before the *sentry-setup* one.

Doing this would leave the config as clean as possible, and may be allow the use of other services.

What do you think?

**Comment by hussfelt** [comment]
Sounds like a plan!

I am pretty lost on how to accomplish that though :)
Are you talking about shipping another initializer with some kind of hook with the ember-cli-sentry package or are you talking about enabling raven-setup initializer to read from another initializer and have a "before" statement in those initializers?

Gosh. :)

I'd love to hack this stuff, I need it asap :) But I need some direction...

**Comment by hussfelt** [comment]
ping @damiencaselli

**Comment by hussfelt** [comment]
@damiencaselli And for the record, I hated the solution above as well - but it was the only way I could pass it down without fiddling too much with the core parts. I just want to get a PoC at this point :)

File: app/initializers/raven-setup.js [hunk 1]
```diff
@@ -14,10 +14,17 @@ export function initialize() {
...
-    release: config.APP.version,
-    whitelistUrls: config.sentry.whitelistUrls
-  }).install();
+  var options = {};
+
+  options.whitelistUrls = config.sentry.whitelistUrls;
+
+  options.release = config.APP.version;
+
+  if (typeof config.sentry.dataCallback === 'function') {
```
Comment (damiencaselli) [inline app/initializers/raven-setup.js]: This is where we want to check service existence.
If it exist (we can assume this is a function for our needs), then data callback is that service.

This also makes the parser much more testable by the way.

**Comment by damiencaselli** [comment]
> Are you talking about shipping another initializer with some kind of hook with the ember-cli-sentry package or are you talking about enabling raven-setup initializer to read from another initializer and have a "before" statement in those initializers?

The latter one hehe

I'll try to POC this as an separate repository to ensure it works and that it doesn't generate ghetto code :smiling_imp:

**Comment by hussfelt** [comment]
@damiencaselli So I'm done? :) It seems you know what you want to do - I just hope that I can get this stuff together! :-)

Tell me if there is anything you need, besides a tester!
Closing this then!

**Comment by hussfelt** [comment]
For the record, I could not get the above solution to work. :)

**Pull request closed without merging**